BrowserKit/Sources/ComponentLibrary/Buttons/PrimaryRoundedButton.swift

```diff
@@ -50,13 +54,21 @@ public class PrimaryRoundedButton: ResizableButton, ThemeApplicable {
         updatedConfiguration.titleTextAttributesTransformer = UIConfigurationTextAttributesTransformer { [weak self] incoming in
         // swiftlint:enable line_length
             var container = incoming
-            container.foregroundColor = self?.foregroundColor
+            if self?.state == .disabled {
+                container.foregroundColor = self?.foregroundColorDisabled
+            } else {
+                container.foregroundColor = self?.foregroundColor
+            }
```

Comment:
On one side I like this because its readable and on another side I feel ternary operator works well for clean approach 
```
container.foregroundColor = self?.state == .disabled ? self?.foregroundColorDisabled : self?.foregroundColor
```

Reply:
agreed, I also prefer ternary. discussed live, but will keep this as is since easier to read